tensorflow_federated/python/core/impl/executors/federating_executor.py

```diff
@@ -479,271 +490,69 @@ def _check_value_compatible_with_placement(self, value, placement, all_equal):
-    py_typecheck.check_type(pre_report.type_signature,
-                            computation_types.FederatedType)
-    type_analysis.check_equivalent_types(pre_report.type_signature.member,
-                                         report_type.parameter)
-
-    report = arg.internal_representation[4]
-    return await self._compute_intrinsic_federated_apply(
-        FederatingExecutorValue(
-            anonymous_tuple.AnonymousTuple([
-                (None, report), (None, pre_report.internal_representation)
-            ]),
-            computation_types.NamedTupleType(
-                (report_type, pre_report.type_signature))))
+  async def _compute_intrinsic_federated_eval_at_clients(self, arg):
+    return await self._strategy.federated_eval_at_clients(arg)
 
   @tracing.trace
   async def _compute_intrinsic_federated_apply(self, arg):
-    return await self._map(arg)
-
-  @tracing.trace
-  async def _compute_intrinsic_federated_broadcast(self, arg):
-    py_typecheck.check_type(arg.internal_representation, list)
-    if len(arg.internal_representation) != 1:
-      raise ValueError(
-          'Federated broadcast expects a value with a single representation, '
-          'found {}.'.format(len(arg.internal_representation)))
-    return await executor_utils.compute_intrinsic_federated_broadcast(self, arg)
-
-  @tracing.trace
-  async def _compute_intrinsic_federated_collect(self, arg):
-    py_typecheck.check_type(arg.type_signature, computation_types.FederatedType)
-    type_analysis.check_federated_type(
-        arg.type_signature, placement=placement_literals.CLIENTS)
-    val = arg.internal_representation
-    py_typecheck.check_type(val, list)
-    member_type = arg.type_signature.member
-    child = self._target_executors[placement_literals.SERVER][0]
-    collected_items = await child.create_value(
-        await asyncio.gather(*[v.compute() for v in val]),
-        computation_types.SequenceType(member_type))
-    return FederatingExecutorValue(
-        [collected_items],
-        computation_types.FederatedType(
-            computation_types.SequenceType(member_type),
-            placement_literals.SERVER,
-            all_equal=True))
+    return await self._strategy.federated_apply(arg)
 
   @tracing.trace
-  async def _compute_intrinsic_federated_eval_at_clients(self, arg):
-    return await self._eval(arg, placement_literals.CLIENTS, False)
+  async def _compute_intrinsic_federated_map(self, arg):
+    return await self._strategy.federated_map(arg)
 
   @tracing.trace
-  async def _compute_intrinsic_federated_eval_at_server(self, arg):
-    return await self._eval(arg, placement_literals.SERVER, True)
+  async def _compute_intrinsic_federated_map_all_equal(self, arg):
+    return await self._strategy.federated_map_all_equal(arg)
 
   @tracing.trace
-  async def _compute_intrinsic_federated_map(self, arg):
-    return await self._map(arg, all_equal=False)
+  async def _compute_intrinsic_federated_broadcast(self, arg):
```

Comment:
I think these additional diffs are happening because I sorted the methods alphabetically since you did this refactor. I think restoring this would reveal that there are actually much less diff that it appears.

Reply:
This makes sense, I'll make that change shortly

Reply:
fixed